subworkflows/local/reference_indexing_multi.nf

```diff
@@ -2,101 +2,202 @@
-                                            .csv
-                                            .splitCsv ( header:true )
+        ch_splitreferencesheet_for_branch = Channel.fromPath(referencesheet)
+                                                .splitCsv ( header:true )
+                                                .map {
+                                                    row ->
+                                                        def meta            = [:]
+                                                        meta.id             = row["reference_name"]
+                                                        def fasta           = file(row["fasta"], checkIfExists: true) // mandatory parameter!
+                                                        def fai             = row["fai"] != "" ? file(row["fai"], checkIfExists: true) : ""
+                                                        def dict            = row["dict"] != "" ? file(row["dict"], checkIfExists: true) : ""
+                                                        def mapper_index    = row["mapper_index"] != "" ? file(row["mapper_index"], checkIfExists: true) : ""
+                                                        def circular_target = row["circular_target"]
+                                                        def mitochondrion   = row["mitochondrion"]
+                                                        [ meta, fasta, fai, dict, mapper_index, circular_target, mitochondrion ]
+                                                }
     }
 
-    // Detect if fasta is gzipped or not, unzip if necessary, and generate meta ID by sanitizing file
-    if ( fasta.extension == 'gz' ) {
-        ch_gz_ref = Channel.fromPath(fasta).map{[[], it]}
-        GUNZIP ( ch_gz_ref )
-        ch_ungz_ref = GUNZIP.out.gunzip.map{[[id: clean_name], it[1] ]}
-    } else {
-        ch_ungz_ref = Channel.fromPath(fasta).map{[[id: clean_name], it ]}
-    }
+    // GENERAL DESCRIPTION FOR NEXT SECTIONS
+    // This will be the same scheme for all other generation steps, i.e.
+    // for those that need to be processed send them to a  multiMap (others skip via branch)
+    // take only those files needed for the generation step in one sub-channel, and all other
+    // channel elements go into 'remainder'. After generation, join these back, and then mix back the
+    // ones that don't skip
+
+    //
+    // DECOMPRESSION
+    //
+
+    // Detect if fasta is gzipped or not
+    ch_fasta_for_gunzip = ch_splitreferencesheet_for_branch
+                            .branch {
+                                meta, fasta, fai, dict, mapper_index, circular_target, mitochondrion ->
+                                    forgunzip: fasta.extension == "gz"
+                                    skip: true
+                            }
+
+    // Pull out name/file to match cardinality for GUNZIP module
+    ch_gunzip_input = ch_fasta_for_gunzip
+        .forgunzip
```

Comment:
I think having the subchannel call on the same line as the channel is more legible (and consistent with rest of code), but as you like.